### PR TITLE
Fix : keep the same order as in the redux store for diagrams

### PR DESCRIPTION
**Pull request by flomillot**

### CODE REVIEW

File: src/components/diagrams/diagram-pane.js [hunk 1]
```diff
@@ -415,9 +415,7 @@ export function DiagramPane({
             if (diagramsToAdd?.length) {
                 // First we add the empty diagrams in the views
                 setViews((views) => {
-                    const updatedViews = views
-                        .concat(diagramsToAdd)
-                        .sort(sortByAlign);
+                    const updatedViews = views.concat(diagramsToAdd);
                     return updatedViews;
```
[the remark this answers is not in the document]
Comment (flomillot) [reply src/components/diagrams/diagram-pane.js:420]: I let it like to have the same syntax as the other setViews, but yeah I can remove this var

File: src/components/diagrams/diagram-utils.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/**
```
[the remark this answers is not in the document]
Comment (flomillot) [reply src/components/diagrams/diagram-utils.js:1]: Yes forgot it, ty

File: src/components/diagrams/diagram-utils.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * @param diagramStates the diagrams array of the redux store
+ * @returns {(function(*, *): (*))|*} new array sorting function based on diagramStates
+ */
+export const sortDiagrams = (diagramStates) => {
+    return (a, b) => {
+        if (a?.align !== b?.align) {
+            return sortByAlign(a, b);
+        } else {
+            return sortByIndex(a, b, diagramStates);
+        }
```
Comment (jonenst) [inline src/components/diagrams/diagram-utils.js:52]: `sortByAlign || sortByIndex` seems to be the canoical way to write this in javascript

Comment (jonenst) [reply src/components/diagrams/diagram-utils.js:52]: also rename sortDiagrams to diagramSorter or something to more easily convey the meaning that you are capturing diagramStates to return a sorting function

File: src/components/diagrams/diagram-utils.test.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { sortDiagrams } from './diagram-utils';
+
+test('diagram-common.sortDiagrams', () => {
```
Comment (jonenst) [inline src/components/diagrams/diagram-utils.test.js:10]: <3 tests

**Review by EstherDarkish — COMMENTED** [review]
I still find cases where this do not work. I'm investigating...

File: src/components/diagrams/diagram-utils.js (hunk 3 above)
Comment (flomillot) [reply src/components/diagrams/diagram-utils.js:52]: Can you write the code suggestion ? I don't see how to write it with ||

File: src/components/diagrams/diagram-utils.test.js (hunk 4 above)
Comment (flomillot) [reply src/components/diagrams/diagram-utils.test.js:10]: It's from Charly tbh :)

**Review by jonenst — APPROVED** [review]

File: src/components/diagrams/diagram-utils.js [hunk 5]
```diff
@@ -30,11 +30,18 @@ const sortByAlign = (a, b) => {
  * Sort by the order (index) of the objects inside diagramStates.
  * So we keep the same order as in the redux store.
  * We use the id of the objects to identify their index.
+ * PS : we have to check the svgType type too because the ID is not unique
```
Comment (EstherDarkish) [inline src/components/diagrams/diagram-utils.js]: "We use the ID and type of the objects to identify their index."

**Review by etiennehomer — APPROVED** [review]

**Review by EstherDarkish — APPROVED** [review]
Code and tests OK

**Pull request merged**